src/nvim/ex_getln.c

```diff
@@ -2428,7 +2427,23 @@ static void abandon_cmdline(void)
 /// @param indent  indent for inside conditionals
 char_u *getcmdline(int firstc, long count, int indent, bool do_concat FUNC_ATTR_UNUSED)
 {
-  return command_line_enter(firstc, count, indent, true);
+  bool check_cmdheight = p_ch < 1 && !ui_has(kUIMessages);
+
+  if (check_cmdheight) {
+    set_option_value("ch", 1L, NULL, 0);
+    redraw_statuslines();
+  }
+
+  char_u *ret = command_line_enter(firstc, count, indent, true);;
+
+  if (check_cmdheight) {
+    set_option_value("ch", 0L, NULL, 0);
+
+    // Buffer redraw is needed if no command line
```

Comment:
This comment looks a bit confusing. Doesn't it redraw everything?

Reply:
OK.  I will improve it later.

Reply:
Oh, I have tested it and the redraw may be not needed.
I will try to remove the redraw.

Reply:
Redrawing everything is suboptimal for this, I'm sure redrawing the current window would be enough? Alternatively, you could check if `redrawcmdline` works, that'd be even better.

Reply:
I have removed the redraw.